pkg/pod/pods.go

```diff
@@ -191,43 +191,38 @@ func getPod(dataDir string, uuid *types.UUID) (*Pod, error) {
 
 	p := &Pod{UUID: uuid, dataDir: dataDir}
 
-	// we try open the pod in all possible directories, in the same order the states occur
-	l, err := lock.NewLock(p.embryoPath(), lock.Dir)
-	if err == nil {
-		p.isEmbryo = true
-	} else if err == lock.ErrNotExist {
-		l, err = lock.NewLock(p.preparePath(), lock.Dir)
+	// dirStates is a list of directories -> state that directory existing
+	// implies.
+	// Its order matches the order states occur.
+	dirStates := []struct {
+		dir           string
+		impliedStates []*bool
+	}{
+		{dir: p.embryoPath(), impliedStates: []*bool{&p.isEmbryo}},
+		// For prepare, assume it's aborted prepare until it gets updated below
+		{dir: p.preparePath(), impliedStates: []*bool{&p.isAbortedPrepare}},
+		{dir: p.preparedPath(), impliedStates: []*bool{&p.isPrepared}},
+		// For run, assume exited until the lock is tested
+		{dir: p.runPath(), impliedStates: []*bool{&p.isExited}},
+		{dir: p.garbagePath(), impliedStates: []*bool{&p.isGarbage}},
```

Comment:
There seems to be a re-ordering happening here between garbage and exitedgarbage. I'm not sure if it is on purpose and if the result is equivalent, but reading the original code I would have expected the last two here to be swapped.